php/utility/utf.php

```diff
@@ -162,11 +162,15 @@ public static function utf8ize($mixed)
 		return($mixed);
 	}
 
+	private static function maybe_mb_convert_to_utf8($string) {
+		return function_exists('mb_convert_encoding') ? mb_convert_encoding($string, "UTF-8", "auto") : $string;
```

Comment:
`function_exists('mb_convert_encoding')` can be a static variable in the UTF class. There is a 0.15s improvement because this method is not cached internally by PHP.

Reply:
> There is a 0.15s improvement because this method is not cached internally by PHP.

`function_exists` is pretty fast compared to `rawurlencode` and `mb_convert_encoding`. Where did you find a significant improvement?

> function_exists('mb_convert_encoding') can be a static variable in the UTF class

I found no significant improvement (when calling `raw_url_encode_static` 5000000 times):
```php
public static function raw_url_encode_static($string) {
  if (Self::$hasMBEncode === -1)
    Self::$hasMBEncode = function_exists('mb_convert_encoding') ? 1 : 0;
  return rawurlencode(Self::$hasMBEncode ? mb_convert_encoding($string, 'UTF-8', "auto") : $string);
}
```
A global appears to be around 10% faster:
```php
public static function raw_url_encode_static($string) {
  global $hasMBEncode;
  return rawurlencode($hasMBEncode ? mb_convert_encoding($string, 'UTF-8', "auto") : $string);
}
```
However, my best guess is that this function is the wrong one to optimize.

Reply:
Okay, thanks for the testing this.  It's not worth the regression potential creating a global over a 10% improvement.